src/libworld/libworld.h

```diff
@@ -2541,6 +2548,138 @@ namespace world
         static chrono::milliseconds getLogTimestamp();
         static void formatLogString(chrono::milliseconds timestamp, char logString[512], const char* format, va_list args);
     };
+
+    class RouteProvider
+    {
+    protected:
+        shared_ptr<HostServices> m_host;
+
+    public:
+        class Route
+        {
+        private:
+            string m_icaoDeparture;
+            string m_icaoDestination;
+            std::vector<string> m_icaoAirframes;
+            string m_AirlineIcao;
+
+        public:
+            const string &departure() const { return m_icaoDeparture; }
+            const string &destination() const { return m_icaoDestination; }
+            const std::vector<string> &usedAirframes() const { return m_icaoAirframes; }
+            const string &airline() const { return m_AirlineIcao; }
+
+        public:
+            Route(const string _departure, const string _destination, const string _airlineIcao, std::vector<string> _airframes) : m_icaoDeparture(_departure),
+                                                                                                                                   m_icaoDestination(_destination),
+                                                                                                                                   m_icaoAirframes(_airframes),
+                                                                                                                                   m_AirlineIcao(_airlineIcao)
+            {
+            }
+        };
+
+    protected:
+        RouteProvider(shared_ptr<HostServices> _host) : m_host(_host){};
+
+    public:
+        virtual const Route &findRandomRouteFrom(const string &fromICAO, const string &airframe = "", const std::vector<std::string> &allowedAirlines = {}) = 0;
+        virtual const Route &findRandomRouteTo(const string &toICAO, const string &airframe = "", const std::vector<std::string> &allowedAirlines = {}) = 0;
+    };
+    class RandomRouteProvider : public RouteProvider
+    {
+    private:
+        vector<shared_ptr<RouteProvider::Route>> m_routes;
+        unordered_map<string, vector<shared_ptr<RouteProvider::Route>>> m_routesFrom;
+
+        unordered_map<string, vector<shared_ptr<RouteProvider::Route>>> m_routesTo;
+
+        mt19937 m_rng;
+
+    public:
+        RandomRouteProvider(shared_ptr<HostServices> _host, vector<shared_ptr<RouteProvider::Route>> _routes) : m_routes(_routes), RouteProvider(_host)
```

Comment:
Isn't it preferred to receive `_routes` as const reference?
```
const vector<shared_ptr<RouteProvider::Route>>& _routes
```
it seems that we create an unnecessary copy here

Reply:
As I said in the previous comment, I am not an expert in C++, I think I used a std::move while calling it to prevent the unnecessary copy. By the way, this part is deprecated.